Python3 version of the archiver

hunter86bg opened the pull request.

hunter86bg commented:
@aravindavk ,
Can you check it out ?

aravindavk commented on archive_georep_changelogs.py:
```diff
@@ -0,0 +1,83 @@
+
+    stime_xattr_values = []
+    for x in xattrs_list:
+        stime = struct.unpack("!II", xattr.getxattr(brick_path, x))
```
This will crash if multiple sessions exist and multiple bricks per node. xattrs_list will have all the session xattrs including session xattr of different volume.

aravindavk commented on archive_georep_changelogs.py:
```diff
@@ -0,0 +1,83 @@
+    """
+    """
+    When removing georep sessions dangling stimes will remain with a value of "0"
+    We want to avoid them , so we check for the stime-xattr-prefix in each gsyncd.conf
```
How about ignoring the xattr values with value `"0"` and use the previous method of xattr list?

```python
if stime[0] > 0:
    stime_xattr_values.append(stime[0])
```

hunter86bg replied:
Never thought about such scenario. It definitely needs better logic.

aravindavk reviewed (commented):
Minor comments. Looks good otherwise.

Replace the existing `main.py` file with this file and update the doc if required.

hunter86bg commented:
@aravindavk ,
can you check the latest version ? The last commit addresses the issues described above.
Instead of searching for stimes in the configs (which could lead to case where we search for stime which is for another volume's session), the script will use the old logic, while avoiding any stale geo-rep sessions' stime (they are always "0x00000000" -> "0").

Let's keep both files ,as there might be still users of EL7 which by default comes with python2.

aravindavk reviewed (approved):
Looks good. Thanks
